core/distributed/matrix.cpp

```diff
@@ -66,15 +66,16 @@ Matrix<ValueType, LocalIndexType, GlobalIndexType>::Matrix(
 template <typename ValueType, typename LocalIndexType, typename GlobalIndexType>
 Matrix<ValueType, LocalIndexType, GlobalIndexType>::Matrix(
     std::shared_ptr<const Executor> exec, mpi::communicator comm,
-    const LinOp* local_matrix_type)
+    pointer_param<const LinOp> local_matrix_type)
```

Comment:
why did I call that one `_type`???
I think using the template suffix is a bit cleaner, like for the other constructor.
```suggestion
    pointer_param<const LinOp> local_matrix_template)
```